### ISSUE TITLE
Error in overload

### ISSUE THREAD

**Issue by ncannasse** [HaxeFoundation/haxe#4803]
I'm now getting extra errors when the JQuery handler get an error inside it:

```
var j = new js.jquery.JQuery();
j.click(function(e) {
    var p = { y : 3 };
    p = { x : 5 };
});

Main.hx:7: lines 7-10 : Could not find a suitable overload, reasons follow
Main.hx:7: lines 7-10 : Overload resolution failed for Void -> js.jquery.JQuery
Main.hx:7: lines 7-10 : Too many arguments
Main.hx:7: lines 7-10 : Overload resolution failed for handler : (js.jquery.Even
t -> Void) -> js.jquery.JQuery
Main.hx:9: characters 8-17 : Object requires field y
Main.hx:9: characters 8-17 : For function argument 'handler'
Main.hx:7: lines 7-10 : Overload resolution failed for ?eventData : Dynamic -> h
andler : (js.jquery.Event -> Void) -> js.jquery.JQuery
Main.hx:9: characters 8-17 : Object requires field y
Main.hx:9: characters 8-17 : For function argument 'handler'
Main.hx:7: lines 7-10 : End of overload failure reasons
```

**Comment by Simn** [comment]
It's supposed to detect that the error is the same for both attempts resulted in the same error. I'll check why that doesn't work here.

**Comment by Simn** [comment]
Oh, I didn't see that there are actually 3 overloads and one of them fails differently. Hmm...

**Comment by Simn** [comment]
The best I can offer here right now is to remove the duplicate error. The problem is that the error message is not exactly wrong: You indeed fail one overload because you have too many arguments and then two others because of a typing error unrelated to the function signature.

That last part is the problem: It's really hard to tell if an error comes from the function signature or not. Before I made this change we missed some such cases, now we get a few too many. This is still the preferable behavior because it only applies to the erroneous cases, instead of blocking of code that's completely valid.

**Comment by ncannasse** [comment]
That's not acceptable for long term in that state. The overload is correct and should not print an error at all. We need to be able to distinguish (using the error position?) if the error comes from the overloading unification or from within the expression.

**Comment by ncannasse** [comment]
Ok of course I didn't see your fix that from 2 years ago :)

**Comment by Simn** [comment]
I only removed the duplicate error message with my fix, but the general issue is covered in #7041. I only linked here because my comment explains why this is so hard to get right.
